x-pack/plugins/maps/public/classes/styles/vector/properties/dynamic_text_property.test.tsx

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+jest.mock('../components/vector_style_editor', () => ({
+  VectorStyleEditor: () => {
+    return <div>mockVectorStyleEditor</div>;
+  },
+}));
+
+import React from 'react';
+
+// @ts-ignore
+import { DynamicTextProperty } from './dynamic_text_property';
+import { RawValue, VECTOR_STYLES } from '../../../../../common/constants';
+import { IField } from '../../../fields/field';
+import { Map as MbMap } from 'mapbox-gl';
+import { mockField, MockLayer, MockStyle } from './test_helpers/test_util';
+import { IVectorLayer } from '../../../layers/vector_layer';
+
+export class MockMbMap {
+  _paintPropertyCalls: unknown[];
+  _lastTextFieldValue: unknown | undefined;
+
+  constructor(lastTextFieldValue?: unknown) {
+    this._paintPropertyCalls = [];
+    this._lastTextFieldValue = lastTextFieldValue;
+  }
+  setLayoutProperty(layerId: string, propName: string, value: undefined | 'string') {
+    if (propName !== 'text-field') {
+      throw new Error('should only use to test `text-field`');
+    }
+    this._lastTextFieldValue = value;
+    this._paintPropertyCalls.push([layerId, value]);
+  }
+
+  getLayoutProperty(layername: string, propName: string): unknown | undefined {
+    if (propName !== 'text-field') {
+      throw new Error('should only use to test `text-field`');
+    }
+    return this._lastTextFieldValue;
+  }
+
+  getPaintPropertyCalls(): unknown[] {
+    return this._paintPropertyCalls;
+  }
+}
+
+const makeProperty = (mockStyle: MockStyle, field: IField | null) => {
+  return new DynamicTextProperty(
+    {},
+    VECTOR_STYLES.LABEL_TEXT,
+    field,
+    (new MockLayer(mockStyle) as unknown) as IVectorLayer,
+    () => {
+      return (value: RawValue) => value + '_format';
+    }
+  );
+};
+
+describe('syncTextFieldWithMb', () => {
```

Comment:
Would it be better to move these tests to StaticTextProperty since that is really the most common scenario where label will be undefined?

Reply:
I added one for that file, iso. just moving it. This PR introduces a brittle change. More test coverage does not hurt in this case imho. e.g. This PR now also introduces a test verifying that the get-lookup for dynamic-tests remains correctly implemented.